### DISCUSSION TITLE
The processed file sounds correctly after the VAD, but the model still seems to know where the big silence gap was and stops at this place.

### DISCUSSION THREAD

**Discussion by ghost** [snakers4/silero-vad#312] [Q&A] [answered]
## ❓Help - The processed file sounds correctly after the VAD, but the model still seems to know where the big silence gap was and stops at this place.

Hi, maybe someone faced with such a situation: I'm trying to apply the silero-VAD before the wav2vec model. The processed file sounds correctly after the VAD, but the model still seems to know where the big silence gap was and stops at this place.
Any guesses as to why this might be?

**Comment by snakers4** [comment]
Not sure I can understand.
Can you maybe post a minimal example?

**Comment by ghost** [comment, marked as answer]
> Not sure I can understand. Can you maybe post a minimal example?

The idea is as follows: apply VAD to the original file _audio.wav_, and use the _collect_chunks()_ function to get the processed file without non-speech fragments - _only_speech.wav_. Then pass this _only_speech.wav_ file as the input to the wav2vec model.
But the thing is, if the original audio file contains longer pauses (about 2 sec), VAD process them successfully and sounds correctly, but the wav2vec model still seems to know where this pause was and stops at this place as if identifying it as the end of the file.
And I can't figure out how the model identifies this pause in the _only_speech.wav_ file. As if there is some marker...

**Reply by snakers4** [reply]
Well, looks like this is an issue of wav2vec, not the vad. We do not put any markers into the file, we just cut it using timestamps.

You can try the following experiment - cut the file a bit differently, i.e. cut extra 0.5s on each silence and see what happens then.

**Reply by ghost** [reply]
yes, the problem was with the wav2vec. I've replaced the wav2vec trained on commonvoice by the wav2vec trained on Librispeech, and it works correctly now.

**Comment by ghost** [comment]
[audio_files.zip](https://github.com/snakers4/silero-vad/files/11021226/audio_files.zip)
there are files for an example.
wav2vec model transcribed just this part (using _only_speech.wav_ file): IN OUR WORK WE ARE OFTEN SURPRISED BY THE FACT THAT MOST PEOPLE KNOW ABOUT AUTOMATIC SPEECH RECOGNITION BUT NOW VERY LITTLE ABOUT VOICE ACTIVITY DETECTION
